### DISCUSSION TITLE
Is there an known issue with VS2022 and Uno 5.1? Have to clean .vs to make the build work

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#15807] [General]
Ever since i upgraded to my uno 5.1, if i make a xaml change i have to delete the .vs directory to make the system build correctly it states that the xaml file i changed is not found during compiilation.

is there something that i setup wrong? this is very frustrating

**Comment by jeromelaban** [comment]
Thanks for the report. The latest visual studio enabled a feature called ["accelerated builds"](https://github.com/dotnet/project-system/blob/main/docs/build-acceleration.md), which can break some apps in some obscure ways, though we may not have found the exact root cause.

Could you try adding this to your `Directory.Build.props` file:

```
<PropertyGroup>
   <AccelerateBuildsInVisualStudio>false</AccelerateBuildsInVisualStudio>
</PropertyGroup>
```

and let us know if it helps?

**Reply by cconner100** [reply]
Thanks,

Tried this but it still faults.  I check all my csproj files to ensure that the AccelerateBuildInVisualStudio is not set and put this flag set to false in the Directory.Build.props file, still have to clean the .vs file for the system to compile correctly.

**Reply by cconner100** [reply]
OK digging more into it, i see in the Uno Platform window the following error

[ERROR] Value cannot be null.
Parameter name: versionRange

any ideas on this?

**Reply by jeromelaban** [reply]
This is likely unrelated and specific to the Uno Dev Server. Can you provide a [binlog of the error](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)? Note that every time there's a build error, we'll ask for one.

**Reply by cconner100** [reply]
here is the binlog
[CentralNode_devenv_PID=8992_x64_BuildManager_Default_0.zip](https://github.com/unoplatform/uno/files/14644045/CentralNode_devenv_PID.8992_x64_BuildManager_Default_0.zip)


The error when i run is
Microsoft.UI.Xaml.Markup.XamlParseException: 'XAML parsing failed.'

if i clean out my .vs directory all is fine after that

Also, if i make a change to the xaml file, sometimes i get a file not found during the build of that xaml file, will send binlog when that happens again

**Reply by cconner100** [reply]
Any thoughts on this, its very frustrating to have to clean the .vs file when i make a change to any xaml file

**Reply by jeromelaban** [reply]
Thanks for the update. At this time, we're not able to reproduce this issue as long as `AccelerateBuildsInVisualStudio` is set to false in the Directory.Build.props. If you're able to reproduce this in a base app, it would help narrow down the issue.
